test/fs/test_idbfs_sync.c

```diff
@@ -67,12 +67,32 @@ void test() {
       result = -11000 - errno;
   }
 
-  // a directory
+  // a directory that we will later recreate from scratch before loading from IDB
   if ((stat("/working1/dir", &st) != -1) || (errno != ENOENT))
     result = -12000 - errno;
   else if (mkdir("/working1/dir", 0777) != 0)
     result = -13000 - errno;
 
+  EM_ASM(
+    setTimeout(() => {
+      // Run the remainder of the test after all already scheduled JavaScript
+      // tasks have executed (such as the one scheduled by IDBFS.queuePersist).
+      // This is to ensure that mkdir also triggers a call to queuePersist and
+      // to isolate mkdir from queuePersist invocations made by other syscalls.
```

Comment:
I'm afraid I still don't understand why this is needed.   Why can't we run the remainder of this test directly?    How can we tell that the mkdir triggers a call to `queuePersist`?

Reply:
Yes I think that makes sense.

What you are trying to do isolute the mkdir so that it doesn't run alongside any other operations.

How about we add another phase to the test.   So you have FIRST and SECOND.   In the SECOND phase you could call *only* mkdir.   After the second phase, without your fix, the mkdir would be lost I assume?

Reply:
> What do you think about making a separate test with two phases each for each write operation that should auto-persist? I think it makes sense since we are trying to verify each write operation's correctness, and it might be nicer than arbitrarily splitting an existing test.

I'm not sure it worth creating that many new tests.. unless we can do it in way that its expressed clearly and concisely.  I'd rather not have N different tests here if we can avoid it.